apis/server/v1alpha2/device_types.go

```diff
@@ -171,9 +175,11 @@ type DeviceParameters struct {
 	// +optional
 	Features map[string]string `json:"features,omitempty"`
 
-	// +immutable
-	// +optional
-	IPAddresses []packngo.IPAddressCreateRequest `json:"ipAddresses,omitempty"`
```

Comment:
`go generate` was choking on this, so I aliased it. The generated documentation and spec relies on xref to the packngo repo. I suspect we don't want that.

https://github.com/packethost/crossplane-provider-equinix-metal/pull/39/files#diff-43c690743c8e782001de93e8166b5888e858256723d702a90bbc8aa8d2e58e47R96

Reply:
```
15:37:39 [ .. ] go generate darwin_amd64
angryjet: error: error loading packages using pattern ./...: /Users/marques/src/crossplane-provider-equinix-metal/apis/server/v1alpha2/zz_generated.deepcopy.go:195:13: (*in)[i].DeepCopyInto undefined (type packngo.IPAddressCreateRequest has no field or method DeepCopyInto)
exit status 1
apis/generate.go:26: running "go": exit status 1
15:37:44 [FAIL]
make: *** [go.generate] Error 1
```

Reply:
The IPAddresses type is now local and copying to packngo.IPAddressCreateRequest is done just before sending the device create EM API call.